### DISCUSSION TITLE
How to Receive Data-Only Messages When FCM Push Notification Permission is Denied

### DISCUSSION THREAD

**Discussion by awais-wego** [invertase/react-native-firebase#8057] [Q&A]
Hey, Team!
Is there a way to still receive FCM Data-Only messages even if the user hasn't allowed permission FCM push notifications for both iOS and android ?

**Comment by mikehardy** [comment]
2 issues here:
1- you want to send notifications to a user that has not allowed notifications? I'm not sure there is any world that works
2- you want to send data-only messages to iOS - that will never have the delivery performance you want (it could be 0% delivery) so will likely not be useful for any important use case

**Reply by awais-wego** [reply]
Hey, @mikehardy !
Thanks for quick reply!

I apologize if my message confused you, but from what I see in this thread https://github.com/invertase/react-native-firebase/discussions/4885?sort=old  
data-only messages are being received on iOS when the app is in the foreground. This is the same behavior I want in my app. However, what happens if the user doesn't grant notification permissions? Are we still able to receive data-only messages in the foreground on iOS ?

From what I understand after reading the thread, we can receive data-only messages on iOS when the app is in the foreground, but is user permission still necessary for this ?



<img width="960" alt="Screenshot 2024-10-11 at 5 56 51 PM" src="https://github.com/user-attachments/assets/9e2fe6b3-58b0-4155-b00b-49bda354350a">

**Reply by mikehardy** [reply]
I'm unsure, you'll have to test it in both cases and report back. I believe it's relatively easy in iOS app settings to toggle notification permission state, and with a test cloud function (or test shell script using the FCM REST APIs) that will send an FCM to a device token, you can make a quick test screen that shows you your device token or calls the function, then send test FCM repeatedly and toggle permission on off to see

Would love to know the results
